app/models/topic_tracking_state.rb

```diff
@@ -151,7 +151,7 @@ def self.publish_unread(post)
 
     group_ids =
       if post.post_type == Post.types[:whisper]
-        [Group::AUTO_GROUPS[:staff]]
+        [Group::AUTO_GROUPS[:staff], *SiteSetting.Whispers.allowed_group_ids]
```

Comment:
If the site setting is blank will this result in nil values? That might cause unintended side effects

Reply:
I think it will be fine because when setting is empty then we have []
```
  class Whispers
    def self.allowed_group_ids
      SiteSetting.whispers_allowed_groups.present? ? SiteSetting.whispers_allowed_groups.split("|").map(&:to_i) : []
    end
  end
```